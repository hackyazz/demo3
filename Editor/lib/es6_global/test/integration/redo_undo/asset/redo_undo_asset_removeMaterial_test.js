

import * as Curry from "../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as RedoUndoTool$WonderEditor from "../tool/RedoUndoTool.js";
import * as ReactTestTool$WonderEditor from "../../../tool/ReactTestTool.js";
import * as BuildComponentTool$WonderEditor from "../../../tool/BuildComponentTool.js";
import * as MainEditorSceneTool$WonderEditor from "../../../tool/MainEditorSceneTool.js";
import * as MainEditorAssetIdTool$WonderEditor from "../../asset/tool/MainEditorAssetIdTool.js";
import * as MainEditorAssetTreeTool$WonderEditor from "../../asset/tool/MainEditorAssetTreeTool.js";
import * as MainEditorAssetHeaderOperateNodeTool$WonderEditor from "../../asset/tool/MainEditorAssetHeaderOperateNodeTool.js";

describe("redo_undo: asset remove material", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        var _simulateOnceRemoveMaterial = function () {
          MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* buildEmptyAssetTree */1](/* () */0);
          var addedMaterialNodeId1 = MainEditorAssetIdTool$WonderEditor.getNewAssetId(undefined, /* () */0);
          MainEditorAssetHeaderOperateNodeTool$WonderEditor.addMaterial(undefined, undefined, /* () */0);
          return MainEditorAssetHeaderOperateNodeTool$WonderEditor.removeMaterialNode(undefined, undefined, addedMaterialNodeId1, /* () */0);
        };
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                MainEditorSceneTool$WonderEditor.initState(sandbox, undefined, undefined, /* () */0);
                return MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorSceneTool$WonderEditor.setFirstCubeToBeCurrentSceneTreeNode);
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        describe("test undo operate", (function () {
                describe("test undo one step", (function () {
                        return Wonder_jest.test("step which from first to zero", (function () {
                                      _simulateOnceRemoveMaterial(/* () */0);
                                      RedoUndoTool$WonderEditor.undoHistoryState(undefined, undefined, undefined, undefined, /* () */0);
                                      return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildAssetChildrenNode(undefined, /* () */0));
                                    }));
                      }));
                return /* () */0;
              }));
        describe("test redo operate", (function () {
                describe("test redo one step", (function () {
                        return Wonder_jest.test("undo step which from first to zero, redo step which from zero to first", (function () {
                                      _simulateOnceRemoveMaterial(/* () */0);
                                      RedoUndoTool$WonderEditor.undoHistoryState(undefined, undefined, undefined, undefined, /* () */0);
                                      RedoUndoTool$WonderEditor.redoHistoryState(undefined, undefined, undefined, undefined, /* () */0);
                                      return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildAssetChildrenNode(undefined, /* () */0));
                                    }));
                      }));
                return /* () */0;
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
