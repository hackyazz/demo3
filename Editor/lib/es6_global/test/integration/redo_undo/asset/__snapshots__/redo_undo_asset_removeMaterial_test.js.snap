// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: asset remove material test redo operate test redo one step undo step which from first to zero, redo step which from zero to first 1`] = `
<article
  className="wonder-asset-assetChildren"
/>
`;

exports[`redo_undo: asset remove material test undo operate test undo one step step which from first to zero 1`] = `
<article
  className="wonder-asset-assetChildren"
>
  <article
    className="wonder-asset-fileBox "
    onClick={[Function]}
  >
    <div
      className="box-image"
    >
      <img
        onDragStart={[Function]}
        src="./public/img/mat.png"
      />
    </div>
    <div
      className="item-text item-active"
    >
      <span>
        New Material
      </span>
    </div>
  </article>
</article>
`;
